server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1095,7 +1096,12 @@ protected Map<String, String> getConfigs() {
     @Override
     public String generateRandomPassword() {
         final Integer passwordLength = vmPasswordLength.value();
-        return PasswordGenerator.generateRandomPassword(passwordLength);
+        String pswd= PasswordGenerator.generateRandomPassword(passwordLength);
```

Comment:
Currently `PasswordGenerator` is only working with alpha numeric characters; therefore, to meet the need you presented, this class should be extended to work with other symbols.

Reply:
so here i have to make a nested class here..?

Reply:
oh nvm i got it sorry for the mess

Reply:
hey @GutoVeronezi  can you check the changed files? 
thankyou

Reply:
i should have used  regex.Pattern and regex.Matcher ? or this works? in other files i saw Pattern  and Matcher being used so should i change it to like that for consistency?